src/vendor/utils/navigation/index.jsx

```diff
@@ -49,6 +49,20 @@ function withNavigation(config) {
         this.updateHistory(change);
       };
 
+      componentDidUpdate(prevProps) {
```

Comment:
You should not need to use `componentDidUpdate()` it is used for notifying some *other* parts of your program that properties changed.  

You must investigate how to detect the back button is clicked so this component knows it should re-render.